### PR TITLE
afstomp: this patch corrects memory-leak in stomp.c; diskq: this ptach corrects use-after-free in dqtool.c

### LINKED ISSUE #3917: Possible memory leak
# syslog-ng
## Version of syslog-ng
3.35.1
## Memory leak
modules/afstomp/stomp.c:105
modules/afstomp/stomp.c:111
##

```
int
stomp_connect(stomp_connection **connection_ref, char *hostname, int port)
{
  stomp_connection *conn;
  conn = g_new0(stomp_connection, 1);  //Memory is allocated
  conn->socket = socket(AF_INET, SOCK_STREAM, 0);
  if (conn->socket == -1)
    {
      msg_error("Failed to create socket!"); // conn is not freed, memory leak
      return FALSE;
    }
  if (!resolve_hostname_to_sockaddr(&conn->remote_sa, AF_INET, hostname))
    {
      msg_error("Failed to resolve hostname in stomp driver", 
                evt_tag_str("hostname", hostname)); // conn is not freed, memory leak

      return FALSE;
    }
  g_sockaddr_set_port(conn->remote_sa, port);
  if (!g_connect(conn->socket, conn->remote_sa))
    {
      msg_error("Stomp connection failed",
                evt_tag_str("host", hostname));
      _stomp_connection_free(conn);
      return FALSE;
    }
  (*connection_ref) = conn;
  return TRUE;
};
```

### PR DESCRIPTION
**Pull request by stasos24**
This patch corrects following bugs: https://github.com/syslog-ng/syslog-ng/issues/3917

Fixes  #3917

### CODE REVIEW

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
"add to whitelist" add author of a Pull Request to whitelist (globally, be careful, it means this user can trigger kira for any PR)
do nothing -> CI won't start)

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
"add to whitelist" add author of a Pull Request to whitelist (globally, be careful, it means this user can trigger kira for any PR)
do nothing -> CI won't start)

**Comment by MrAnno** [comment]
@kira-syslogng ok to test

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by MrAnno** [comment]
Thanks for opening this PR. We'll review it soon.

About the CI failures:
- commit-check: your first commit message needs to be reformatted a bit, and we require our contributors to sign off their commits
- style-check: the `exit` label should not be aligned at all. It looks weird at first, I can only agree. If you install `astyle` in your dev environment the `make style-format` command will format the code for you.

These can be fixed easily by an interactive git-rebase. Let us know if we can help with that.

----

Some additional notes:

Please remove `badly-formatted-files.list` from your second commit.
I think it would be cleaner to add each fix to a separate commit, for example:
- commit 1:
  ```
  afstomp: fix memory leaks

  Signed-off-by: Stanislav ...
  ```
- commit 2:
  ```
  dqtool: correct use-after-free during relocation

  Signed-off-by: Stanislav ...
  ```

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: modules/afstomp/stomp.c [hunk 1]
```diff
@@ -104,14 +104,15 @@ stomp_connect(stomp_connection **connection_ref, char *hostname, int port)
...
+      g_free(conn);
       return FALSE;
     }
 
   if (!resolve_hostname_to_sockaddr(&conn->remote_sa, AF_INET, hostname))
     {
       msg_error("Failed to resolve hostname in stomp driver",
                 evt_tag_str("hostname", hostname));
-
+      _stomp_connection_free(conn);
```
Comment (MrAnno) [inline modules/afstomp/stomp.c:115]: `g_sockaddr_unref()` is NULL-safe. 👍🏻

**Review by MrAnno — APPROVED** [review]
Nice catch.

**Review by Kokan — APPROVED** [review]

**Pull request merged**